scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -33,7 +33,7 @@ import scala.util.Failure
 
 class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   require(Set("", EOL).contains(dialect.toplevelSeparator))
-  private val soft = new SoftKeywords(dialect)
+  private val soft = new SoftKeywords(dialect, (token: Token) => token.next)
```

Comment:
next is needed to make sure that `[` or `(` is actually next. Might be problematic to use it in scalafmt, but I think doable? @olafurpg